Various cleanups to ir_emitter and friends

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46686 Various cleanups to ir_emitter and friends**

I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

Differential Revision: [D24464197](https://our.internmc.facebook.com/intern/diff/D24464197)

suo committed:
Various cleanups to ir_emitter and friends

I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

[ghstack-poisoned]

suo committed:
Update on "Various cleanups to ir_emitter and friends"


I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

[ghstack-poisoned]

suo committed:
Update on "Various cleanups to ir_emitter and friends"


I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

Differential Revision: [D24464197](https://our.internmc.facebook.com/intern/diff/D24464197)

[ghstack-poisoned]

suo committed:
Update on "Various cleanups to ir_emitter and friends"


I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

Differential Revision: [D24464197](https://our.internmc.facebook.com/intern/diff/D24464197)

[ghstack-poisoned]

SplitInfinity commented on torch/csrc/jit/frontend/ir_emitter.cpp:
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/frontend/ir_emitter.h>
+
```
Intentional?

SplitInfinity commented on torch/csrc/jit/frontend/sugared_value.cpp:
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/frontend/sugared_value.h>
+
```
Intentional?

SplitInfinity commented on torch/csrc/jit/python/python_sugared_value.cpp:
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/python/python_sugared_value.h>
+
```
Intentional?

suo replied:
yes, otherwise my clang-format reorders the includes alphabetically

SplitInfinity commented on torch/csrc/jit/python/python_sugared_value.cpp:
```diff
@@ -688,28 +688,28 @@ std::shared_ptr<SugaredValue> BooleanDispatchValue::call(
+    message_values.reserve(args.size() + kwargs.size());
 
-    for (auto inp : inputs) {
+    for (auto inp : args) {
```
```suggestion
    for (const auto& inp : args) {
```

SplitInfinity commented on torch/csrc/jit/python/python_sugared_value.cpp:
```diff
@@ -688,28 +688,28 @@ std::shared_ptr<SugaredValue> BooleanDispatchValue::call(
       message_values.push_back(inp.value(*caller.graph()));
     }
-    for (auto kwarg_inp : attributes) {
+    for (auto kwarg_inp : kwargs) {
```
```suggestion
    for (const auto& kwarg_inp : kwargs) {
```

SplitInfinity reviewed (approved):
🚀

eellison commented on aten/src/ATen/core/interned_strings.h:
```diff
@@ -55,7 +55,7 @@ namespace c10 {
   _(prim, BreakStmt)                 \
   _(prim, ContinueStmt)              \
-  _(prim, LocalVariableScope)        \
+  _(prim, ListComprehension)         \
```
nit: ListComprehension -> ListComprehensionScope. it's not literally a ListComprehension, it's the scope that's created by one

eellison reviewed (commented):
looks good!

suo committed:
Update on "Various cleanups to ir_emitter and friends"


I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

Differential Revision: [D24464197](https://our.internmc.facebook.com/intern/diff/D24464197)

[ghstack-poisoned]

suo committed:
Update on "Various cleanups to ir_emitter and friends"


I was trying to page this code back in after a while and some things
stuck out as unnecessarily confusing.

1. Improve documentation of closures and fork stuff to be more accurate
to how we use them today.
2. Change `prim::LocalVariableScope` to `prim::ListComprehension`. It is
only ever used for a list comprehensions, and in general the nodes
emitted by `ir_emitter` should correspond to concrete operations or
language features rather than semantic constraints.
3. Change the somewhat mysterious "inputs" and "attributes" argument
names throughout the codebase to be the more obvious "args" and "kwargs"
that they generally represent (I think "inputs" and "attributes" come
from the AST naming).

Differential Revision: [D24464197](https://our.internmc.facebook.com/intern/diff/D24464197)

[ghstack-poisoned]
